sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py

```diff
@@ -1256,14 +1257,24 @@ def _generate_delete_blobs_options(self,
                     if_tags_match_condition=if_tags_match_condition
                 )
 
-            query_parameters, header_parameters = self._generate_delete_blobs_subrequest_options(**options)
+            built_url = "/{}/{}{}".format(quote(container_name), quote(blob_name, safe='/~'), self._query_str)
 
-            req = HttpRequest(
-                "DELETE",
-                "/{}/{}{}".format(quote(container_name), quote(blob_name, safe='/~'), self._query_str),
-                headers=header_parameters
-            )
-            req.format_parameters(query_parameters)
+            # TODO: Shouldn't need to POP these
+            # Should just be not included or done better somehow
+            if 'etag' in options:
+                options.pop('etag')
+            if 'lease_access_conditions' in options:
+                options.pop('lease_access_conditions')
+            if 'modified_access_conditions' in options:
+                options.pop('modified_access_conditions')
+
+            req = build_delete_request(built_url, **options)
+            if 'x-ms-version' in req.headers:
+                req.headers.pop('x-ms-version')
+            if '?' in req.url:
```

Comment:
`build_delete_request` was popping in a delete type of `permanent` causing Storage to kickback saying permanent deletion is not enabled, so have to strip after the `?`